### PR TITLE
Add option to convert to unscanned ckpt

### PR DESCRIPTION
**Pull request by RissyRan**
# Description

Add option to convert to unscanned ckpt directly, instead of converting from source to scanned, then to unscanned version

# Tests

Run conversion for V3 and V2-Lite:
* V3 tested in debugging
* V2-Lite tested with same decoding results `Input I love to play ->  with my food. I love to make it` with ckpt converted from scanned version

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [x] I have performed a self-review of my code.
- [x] I have necessary comments in my code, particularly in hard-to-understand areas.
- [x] I have run end-to-end tests tests and provided workload links above if applicable.
- [x] I have made or will make corresponding changes to the doc if needed.

### CODE REVIEW

File: end_to_end/tpu/deepseek/v3-671b/test_deepseek.sh [hunk 1]
```diff
@@ -26,18 +26,18 @@ export MODEL_BUCKET=gs://maxtext-deepseek/deepseek3-671b
 JAX_PLATFORMS=cpu python MaxText/convert_deepseek_ckpt.py.py --base_model_path ${CHKPT_BUCKET} --maxtext_model_path ${MODEL_BUCKET}/${idx} --model_size ${MODEL_VARIATION}
 
 # Step 2:
+# Note that the `CONVERTED_CHECKPOINT` is in a `scanned` format which is great for training but for efficient decoding performance we want the checkpoint in an `unscanned` format.
```
Comment (mitalisi) [inline end_to_end/tpu/deepseek/v3-671b/test_deepseek.sh:29]: How do we verify that the conversion is correct? Is there a logits test we could add here?

Comment (RissyRan) [reply end_to_end/tpu/deepseek/v3-671b/test_deepseek.sh:29]: Good question! We added this for V2, and planning to do so for V3 as well. That's what we are testing right now. We have seen reasonable tokens outputs, but logits-wise still have some delta.

Comment (mitalisi) [reply end_to_end/tpu/deepseek/v3-671b/test_deepseek.sh:29]: Lets add the test script here as part of this test. We can run it for R1 as well.

**Review by gagika — APPROVED** [review]

File: end_to_end/tpu/deepseek/v3-671b/test_deepseek.sh (hunk 1 above)
Comment (RissyRan) [reply end_to_end/tpu/deepseek/v3-671b/test_deepseek.sh:28]: Discussed offline, we will follow it up.
